### PR TITLE
MAINT: remove redundant code in `MaskedArray.__new__`

### PR DESCRIPTION
**Pull request by eric-wieser**
Trivial cleanups found while investigating the `shrink` argument

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -2875,7 +2868,7 @@ def __new__(cls, data=None, mask=nomask, dtype=None, copy=False,
                     _data._mask = mask
                     _data._sharedmask = not copy
                 else:
-                    if names_:
+                    if _data.dtype.names:
```
Comment (eric-wieser) [inline numpy/ma/core.py]: This should be `is not None`, but that's coming in a future PR

File: numpy/ma/core.py [hunk 2]
```diff
@@ -2884,7 +2877,6 @@ def _recursive_or(a, b):
                                     _recursive_or(af, bf)
                                 else:
                                     af |= bf
-                            return
```
Comment (mhvk) [inline numpy/ma/core.py]: Maybe leave an open line

**Review by mhvk — APPROVED** [review]
Cannot seem to find anything but the ultimate nitpick.

File: numpy/ma/core.py (hunk 2 above)
Comment (eric-wieser) [reply numpy/ma/core.py]: Done

**Pull request merged**